rpc/client.go

```diff
@@ -265,18 +265,19 @@ func (c *Client) heartbeat() error {
 				remoteTimeNow := response.ServerTime + (receiveTime-sendTime)/2
 				c.offset.Offset = remoteTimeNow - receiveTime
 			}
+			offset := c.offset
 			c.mu.Unlock()
-			c.remoteClocks.UpdateOffset(c.addr.String(), c.offset)
+			c.remoteClocks.UpdateOffset(c.addr.String(), offset)
```

Comment:
Can we rearrange this so that we only call UpdateOffset for valid (finite) offsets? It looks like we could probably remove proto.InfiniteOffset entirely.